vdr/src/client/client.rs

```diff
@@ -484,6 +593,74 @@ pub mod test {
             assert_eq!(error, expected_error);
         }
 
+        #[rstest]
+        #[case::regular_error(
+            "0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001a4e6f7420656e6f7567682045746865722070726f76696465642e000000000000", 
+            VdrError::ClientTransactionReverted("Error(message: Not enough Ether provided.)".to_string()),
+        )]
+        #[case::panic_error(
+            "0x4e487b710000000000000000000000000000000000000000000000000000000000000011",
+            VdrError::ClientTransactionReverted("Panic(code: 11)".to_string()),
+        )]
+        #[case::custom_error(
+            "0x863b93fe000000000000000000000000f0e2db6c8dc6c681bb5d6ad121a107f300e9b2b5",
+            VdrError::ClientTransactionReverted("DidNotFound(identity: f0e2db6c8dc6c681bb5d6ad121a107f300e9b2b5)".to_string()),
+        )]
+        #[case::error_without_required_argument(
+            "0x863b93fe",
+            VdrError::ContractInvalidResponseData("Unable to parse the revert reason '0x863b93fe': Failed to decode the arguments".to_string()),
+        )]
+        #[case::error_with_extra_argument(
+            "0x4e487b71000000000000000000000000000000000000000000000000000000000000001100000000000000000000000000000000000000000000000000000000000011",
+            VdrError::ClientTransactionReverted("Panic(code: 11)".to_string()),
```

Comment:
Don't we want to parse panic codes?

Reply:
It would be good to implement, it will be implemented in follow up task